torchtext/datasets/text_classification.py

```diff
@@ -373,6 +408,122 @@ def AmazonReviewFull(*args, **kwargs):
+                for tokens in _iter:
+                    _data.append((0,
+                                  torch.tensor([token_id for token_id in tokens]).long()))
+    return _data
+
+
+def IMDB(tokenizer=get_tokenizer("basic_english"),
+         root='.data', vocab=None, removed_tokens=[],
+         data_select=('train', 'test')):
+
+    """ Defines IMDB datasets.
+        The labels includes:
+            - 0 : Negative
+            - 1 : Positive
+
+    Create sentiment analysis dataset: IMDB
+
+    Separately returns the training and test dataset
+
+    Arguments:
+        tokenizer: the tokenizer used to preprocess raw text data.
+            The default one is basic_english tokenizer in fastText. spacy tokenizer
+            is supported as well (see example below). A custom tokenizer is callable
+            function with input of a string and output of a token list.
+        root: Directory where the datasets are saved. Default: ".data"
+        vocab: Vocabulary used for dataset. If None, it will generate a new
+            vocabulary based on the train data set.
+        removed_tokens: removed tokens from output dataset (Default: [])
+        data_select: a string or tupel for the returned datasets
+            (Default: ('train', 'test'))
+            By default, all the three datasets (train, test) are generated. Users
+            could also choose any one of them, for example ('test') or
+            just a string 'train'. If 'train' is not in the tuple or string, a vocab
+            object should be provided which will be used to process valid and/or test
+            data.
+
+    Examples:
+        >>> from torchtext.datasets import IMDB
+        >>> from torchtext.data.utils import get_tokenizer
+        >>> tokenizer = get_tokenizer("spacy")
+        >>> train_dataset, test_dataset = IMDB(tokenizer=tokenizer)
+        >>> vocab = train_dataset.get_vocab()
+        >>> train_dataset, = IMDB(tokenizer=tokenizer, vocab=vocab,
+                                  data_select='train')
+        >>>
+        >>> # Attach text pre-processing iterator to tokenizer
+        >>> # For example, add ngrams iterator to tokenizer
+        >>> from torchtext.data.utils import ngrams_iterator
+        >>> def ngrams_tokenizer_wrapper(ngrams):
+        >>>     spacy_tokenizer = get_tokenizer("spacy")
+        >>>     def _wrapper(input_txt):
+        >>>         txt = ngrams_iterator(spacy_tokenizer(input_txt),
+        >>>                               ngrams)
+        >>>         return list(txt)
+        >>>     return _wrapper
+        >>> tokenizer = ngrams_tokenizer_wrapper(2)
+        >>> train_dataset, test_dataset = IMDB(tokenizer=tokenizer)
+    """
+
+    dataset_tar = download_from_url(URLS['IMDB'], root=root)
+    extracted_files = extract_archive(dataset_tar)
+
+    if isinstance(data_select, str):
+        data_select = [data_select]
+    for key in data_select:
```

Comment:
Does this mechanism exist for the other functions? Looks like an extension that is only supported for IMDB.

Reply:
Yes. I extended it to all the other datasets.